src/openms/include/OpenMS/ANALYSIS/XLMS/OpenProXLUtils.h

```diff
@@ -0,0 +1,302 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL ANY OF THE AUTHORS OR THE CONTRIBUTING
+// INSTITUTIONS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR
+// OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+// ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// --------------------------------------------------------------------------
+// $Maintainer: Eugen Netz $
+// $Authors: Eugen Netz $
+// --------------------------------------------------------------------------
+
+#ifndef OPENMS_ANALYSIS_XLMS_OPENPROXLUTILS
+#define OPENMS_ANALYSIS_XLMS_OPENPROXLUTILS
+
+#include <OpenMS/KERNEL/StandardTypes.h>
+#include <OpenMS/CHEMISTRY/ResidueModification.h>
+#include <OpenMS/FORMAT/FASTAFile.h>
+#include <OpenMS/CHEMISTRY/EnzymaticDigestion.h>
+#include <numeric>
+
+namespace OpenMS
+{
+
+  struct ProteinProteinCrossLink
+    {
+      /** Enums
+      */
+      //@{
+      /** @brief type of Protein-Protein cross-link
+      */
+      enum ProteinProteinCrossLinkType
+      {
+        CROSS = 0,
+        MONO = 1,
+        LOOP = 2,
+        NUMBER_OF_CROSS_LINK_TYPES
+      };
+
+      AASequence alpha; // longer peptide
+      AASequence beta; // shorter peptide (empty for mono-link), tie bracker: mass then lexicographical
+      std::pair<SignedSize, SignedSize> cross_link_position; // index in alpha, beta or between alpha, alpha in loop-links
+      double cross_linker_mass;
+      String cross_linker_name;
+      ResidueModification::TermSpecificity term_spec_alpha;
+      ResidueModification::TermSpecificity term_spec_beta;
+
+      ProteinProteinCrossLinkType getType() const
+      {
+        if (!beta.empty()) return CROSS;
+
+        if (cross_link_position.second == -1) return MONO;
+
+        return LOOP;
+      }
+
+    };
+
+struct CrossLinkSpectrumMatch
+  {
+    /// structure of the cross-link
+    ProteinProteinCrossLink cross_link;
+
+    /// reference to pair of spectra
+    Size scan_index_light;
+    Size scan_index_heavy;
+
+    /// final score
+    double score;
+
+    /// rank among the matches to the same spectrum
+    Size rank;
+
+    /// counts, scores and other data for xQuest-like output
+    double pre_score;
+    double percTIC;
+    double wTIC;
+    double int_sum;
+    double match_odds;
+    std::vector< double > xcorrx;
+    double xcorrx_max;
+    std::vector< double > xcorrc;
+    double xcorrc_max;
+    Size matched_common_alpha;
+    Size matched_common_beta;
+    Size matched_xlink_alpha;
+    Size matched_xlink_beta;
+    double HyperCommon;
+    double HyperXlink;
+    double HyperAlpha;
```

Comment:
some of these scores seem to be used only for testing, e.g. `HyperAlpha` so they should not be not be in the final release. I suggest to use a DEBUG_XLINK preprocessor flag to turn on and off some of the testing stuff.